packages/teleport-generator-core/src/validator/index.ts

```diff
@@ -38,6 +42,66 @@ export default class Validator {
 
     return { valid: true, errorMsg: '' }
   }
+
+  public validateComponentContent(input: any): ValidationResult {
+    const verifyProps = checkContent(input, 'prop')
```

Comment:
can you check both prop and state into a single pass? might be a micro-optimization, but when we validate a project of 1000 components in theory, it would mean a lot of passes through the UIDL with traverse

Reply:
also, I would give the functions a more suggestive name, "checkDynamicDefinitions"?